codespell_lib/_codespell.py

```diff
@@ -28,6 +28,8 @@
 import textwrap
 
 word_regex_def = u"[\\w\\-'’`]+"
+# Matches common URIs and email addresses, in that order.
+ignore_word_regex_def = r"(?:(?:https?|ftp|smtp):\/\/([\w-]+\.)+\w{2,}(?:/(?:[\w:/?#\[\]@!$&'()*+,;=.~-]*/?)*)?|[\w.%+-]+@[\w.-]+\.[a-z]{2,})"  # noqa: E501
```

Comment:
Perhaps add file/git/smb (and tftp as it's an easy addition on the existing ftp).

SMTP isn't listed here and I don't think I've particularly seen it either.
https://en.wikipedia.org/wiki/List_of_URI_schemes

Reply:
Also you should optionally accept `mailto:` at the start of the email regex.

Reply:
This doesn't look like it will accept an IPv4 address (or probably an IPv6 one), which it should ideally match too. Was it sourced from somewhere or did you write it yourself?

Reply:
I'll keep this in mind for a possible uri ignore, but it's removed from this PR.

This is something I wrote myself, just trying to write a short/readable matcher, vs something that'd only match canonical HTTP; I don't think even the examples I found were doing IPv6. I looked to RFCs for the path/cgi/anchor char list. If you have a regex you'd prefer, please let me know.

Reply:
I've not looked, but I assume there should be some available somewhere which have been well tested and cover a lot of edge cases.

Reply:
Resolving from this PR anyway.